### DISCUSSION TITLE
[2022/04/26] Frontend Night

### DISCUSSION THREAD

**Discussion by jjenko** [wantedly/frontend-night#202] [Archive]
## Policy

- 資料の準備をがんばらない
- 参加・途中退席は自由
- スキップしない

話したい・聞きたいネタを書いてく ✏️

ハッシュタグ [#frontend_night](https://twitter.com/hashtag/frontend_night?src=hash)

Prev: https://github.com/wantedly/frontend-night/discussions/201 

 _Created from https://github.com/wantedly/frontend-night/discussions/179 by [issue-creator](https://github.com/rerost/issue-creator)_

**Comment by kobayang** [comment]
## [Jest 28: Shedding weight and improving compatibility 🫶](https://jestjs.io/blog/2022/04/25/jest-28)

> Jest now includes a new [--shard](https://jestjs.io/docs/cli#--shard) CLI option, contributed by [Mario Nebl](https://github.com/marionebl). It allows you to run parts of your test across different machine, and has been one of Jest's oldest feature requests.
> Jest's own test suite on CI went from about 10 minutes to 3 on Ubuntu, and on Windows from 20 minutes to 7.

`--shard` option でテストを分割実行できるようになる？早くなるらしい。

**Comment by kobayang** [comment]
## [フロントエンドのテスト戦略について考える](https://zenn.dev/koki_tech/articles/a96e58695540a7)

- テストを書くかどうかは信頼性と実装コスト&実行速度とのトレードオフである
- [テストトロフィー](https://testingjavascript.com/)の概念。
  - <img width="360px" src="https://user-images.githubusercontent.com/12181179/165205089-bc6703fc-48d4-4613-b6c5-995400733827.png" />
- 記事で紹介しているテスト戦略

> 1. トレードオフの観点でバランスのよい結合テストを厚めに書く
> 2. E2E テストは、課金導線やタイムラインなどの、不具合が発生するとビジネス上のネガティブインパクトの大きい箇所だけ書く
> 3. 単体テストは、明らかにテストしなくても自明なロジックに対しては書かない。複雑性が高いビジネスロジックの関数に関しては書く
> 4. 静的テストはベースラインとして必ず引く。導入が後回しになればなるほど導入コストが跳ね上がるので、プロジェクトの最初に必ず入れる

**Comment by kobayang** [comment]
[Navigation API による「JS での画面遷移」と SPA の改善](https://blog.jxck.io/entries/2022-04-22/navigation-api.html)

> - 従来の History API を改善する Navigation API の仕様策定と実装が進んでいる。
> - History API の使いにくかった部分を補うだけではなく、「JS で画面遷移をする」という現状のミッシングピースに取り組み、 SPA が抱える多くの問題だけでなく MPA すら改善する可能性がある。


> SPA として質の高い UX を実装したと思っている開発者を尻目に、支援技術を通じてアクセスしているユーザには、使いにくい実装が多々ある。これは、ある一定の「課題の解決」はしたが、「問題の解決」には至ってない典型例といえる。
> 
> この根本的な問題は、 SPA を表示する際にブラウザが知るべき情報を、ブラウザに提供するためのセマンティクスが不足していたことにある。ここをカバーするために提案されたのが Navigation API だ。

Navigation API spec: https://wicg.github.io/navigation-api/
（まだChromeでIntent to Shipが出たレベルなので実務で使えるのはまだ先）

**Reply by kobayang** [reply]
- Current Origin の履歴の一覧が取得できる
  - ページの中に「戻る」ボタンが置けるようになる
- 画面の遷移処理を記述できる
  - MPAをSPA Likeにできるかも
  - SPA で遷移後のフォーカス・スクロールの制御ができるようになる
- 遷移の開始と終了が取得できるようになる

**Comment by izumin5210** [comment]
電子情報学特論：Chromiumのアーキテクチャを解き明かす
https://docs.google.com/presentation/d/12wd3hLkXVny0b5LnzizmH_3xe8zJ2WY5_9JprfIkp-o/edit?usp=sharing

**Comment by kobayang** [comment]
https://zenn.dev/stin/articles/getting-started-with-lexical
